Auto versioning update

hadijannat opened the pull request:
This PR updates pyproject.toml to use setuptools_scm for automated versioning:

- Removes static version
- Adds `dynamic = ["version"]` to `[project]` section
- Configures setuptools_scm

This change allows automatic version management based on Git tags, addressing issue #303.

To use:
1. Create Git tag: `git tag v1.0.1`
2. Push tag: `git push --tags`

Versions between releases will be like `1.0.1.dev4+g12345`.

Please review and suggest any needed changes.

hadijannat committed:
Migrate to pyproject.toml and update related files

hadijannat committed:
Update pyproject.toml and setup.py based on review comments
- Remove reference to aas-compliance-check from pyproject.toml
- Add newline at end of pyproject.toml
- Remove entry point for aas-compliance-check from setup.py

hadijannat committed:
Update pyproject.toml and setup.py

hadijannat committed:
Add dev extras to setup.py

hadijannat committed:
Add dev optional dependencies to pyproject.toml

hadijannat committed:
Address PR comments: Update pyproject.toml and CI workflow

hadijannat committed:
Address PR comments: Update pyproject.toml and CI workflow

hadijannat committed:
Implement automated versioning using setuptools_scm

hadijannat committed:
Merge branch 'main' into auto-versioning-update

s-heppner commented on .gitignore:
```diff
@@ -25,3 +25,26 @@
 /test/adapter/schemas
+
+
+# Python build artifacts
+__pycache__/
+*.py[cod]
+*$py.class
+*.so
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+*.egg-info/
+.installed.cfg
+*.egg
```
Those are all common Python build artifacts. It would be cleaner to only mention those that actually get created with our build process, which depends on how we actually end up building the module.

s-heppner commented on pyproject.toml:
```diff
@@ -24,6 +23,12 @@ dependencies = [
     "urllib3>=1.26,<3",
     "pyecma376-2>=0.2.4"
 ]
+# This line indicates that the version is dynamically determined
```
This comment doesn't say anything about how the dynamic versioning works, rather it states what is obvious from reading the code. 
 
It'd be better the explain here that we use `setuptools_scm` to dynamically determine the version based on the git tag, as you explain in the PR.

s-heppner commented on pyproject.toml:
```diff
@@ -24,6 +23,12 @@ dependencies = [
+dynamic = ["version"]
+
+[tool.setuptools_scm]
+# This section configures setuptools_scm
```
This comment is obvious as well. 
Please add comments to what's the idea behind the configuration and what it achieves.

s-heppner reviewed (changes requested):
Other than the documentation (comments), it LGTM

hadijannat committed:
Update .gitignore

Clean up .gitignore to include only relevant build artifacts

hadijannat committed:
Update pyproject.toml

Added a more detailed comment for the setuptools_scm configuration, explaining its purpose and behavior.

s-heppner commented on .gitignore:
```diff
@@ -1,27 +1,24 @@
-# Python caching
-**/__pycache__/
-*.pyc
-/.mypy_cache/
```
I'd reintroduce the `/.mypy_cache/` since we're actively using mypy in our CI

s-heppner commented on .gitignore:
```diff
@@ -1,27 +1,24 @@
+*.bak
+*.swp
 
-# Coverage artifacts
-/.coverage
-/htmlcov/
-/docs/build/
-/.hypothesis/
```
Why were these removed? We're using coverage and the sphinx documentation files are in `/docs/build/`.

s-heppner commented on .gitignore:
```diff
@@ -1,27 +1,24 @@
-/docs/build/
-/.hypothesis/
-
-# customized config files
-/test/test_config.ini
```
I would keep this

s-heppner commented on .gitignore:
```diff
@@ -1,27 +1,24 @@
-# customized config files
-/test/test_config.ini
-# Schema files needed for testing
-/test/adapter/schemas
```
These are also actively used in the CI, so it makes sense to keep them

s-heppner commented:
The issues have been solved in #313, therefore this PR is not needed anymore.
